### PR TITLE
PP-4650 Refactoring to prepare for addition of Google Pay.

### PR DESCRIPTION
**Pull request by danworth**
- Moved `applepayment` package into new `wallets` package.
- Rename `ApplePaymentInfo` to `PaymentInfo` and move into `wallets/common/api` package
to be shared with Google Pay.
- Change url for apple pay to `/wallets/apple-pay`.
- Changed name of `ApplePayToken` to `ApplePayAuthRequest` to make it clear
it represents a request which can be validated.

## WHAT
Some of the existing Apple Pay code can be re-used for Google Pay. This is a little refactoring to prepare for a new Google pay end-point and auth request class to following in a separate PR.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/wallets/applepay/WalletType.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package uk.gov.pay.connector.wallets.applepay;
```
Comment (georgeracu) [inline src/main/java/uk/gov/pay/connector/wallets/applepay/WalletType.java]: I think this class can go in `wallets` package as it contains types of wallets, not types of applepay wallets.

File: src/main/java/uk/gov/pay/connector/wallets/common/api/PaymentInfo.java [hunk 2]
```diff
@@ -1,21 +1,22 @@
-package uk.gov.pay.connector.applepay.api;
+package uk.gov.pay.connector.wallets.common.api;
 
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
 import com.fasterxml.jackson.databind.annotation.JsonNaming;
 import uk.gov.pay.connector.gateway.model.PayersCardType;
 
 @JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
-public class ApplePaymentInfo {
+public class PaymentInfo {
```
Comment (georgeracu) [inline src/main/java/uk/gov/pay/connector/wallets/common/api/PaymentInfo.java]: I wonder if `WalletPaymentInfo` explains better what this class is intended to used for. We might confuse people with `PaymentInfo` which is generic and could be reused in `card` payments.

File: src/test/java/uk/gov/pay/connector/it/base/ChargingITestBase.java [hunk 3]
```diff
@@ -228,8 +228,8 @@ protected void shouldReturnErrorForAuthorisationDetailsWithMessage(String author
         assertFrontendChargeStatusIs(chargeId, status);
     }
 
-    public static String authoriseChargeUrlForWallet(String chargeId) {
-        return "/v1/frontend/charges/{chargeId}/wallets".replace("{chargeId}", chargeId);
+    public static String authoriseChargeUrlForApplePay(String chargeId) {
+        return "/v1/frontend/charges/{chargeId}/wallets/apple-pay".replace("{chargeId}", chargeId);
```
Comment (georgeracu) [inline src/test/java/uk/gov/pay/connector/it/base/ChargingITestBase.java]: It really needs the `-pay` part? We will just repeat ourselves when other wallets will be added.

File: src/main/java/uk/gov/pay/connector/wallets/common/api/PaymentInfo.java [hunk 4]
```diff
@@ -1,21 +1,22 @@
-package uk.gov.pay.connector.applepay.api;
+package uk.gov.pay.connector.wallets.common.api;
```
Comment (georgeracu) [inline src/main/java/uk/gov/pay/connector/wallets/common/api/PaymentInfo.java]: I would argue on two things here. First is the use of `api` as we reuse this object to send it for gateway authorisation, so it goes a few levels down into our stack, doesn't stop at the api edge.
Second is the use of `common`. I believe that everything that is not in any of the specific wallet type packages is considered to be common. I would get rid of the `common` part from this package name.

**Review by georgeracu — CHANGES_REQUESTED** [review]
Overall a good separation of the domain concepts with explicit names. 

The only pain points I found were related to the old problem of naming `common` or `util` things. This usually creates a generic place where we will throw everything that doesn't fit everywhere else. 
In this particular case we can get rid of the `common` package and keep only `api`. I believe classes in the parent package are implied to be `common` to the specialised types. This way `...wallets.api.WalletPaymentInfo` will tell us a few things: 
* when is used in `applepay` then we understand that this is a common class because it is in the parent package; 
* we will force people in the future to name better their common packages, because there is no `common` package. We basically avoid creating a `kitchen sink` in our code;
* we will structure `wallets` as if it wasn't a bag of utilities for the wallet types we use. It will have a clean structure explaining its functionality.

File: src/main/java/uk/gov/pay/connector/wallets/applepay/WalletType.java (hunk 1 above)
Comment (danworth) [reply src/main/java/uk/gov/pay/connector/wallets/applepay/WalletType.java]: 👍 good idea

File: src/main/java/uk/gov/pay/connector/wallets/common/api/PaymentInfo.java (hunk 2 above)
Comment (danworth) [reply src/main/java/uk/gov/pay/connector/wallets/common/api/PaymentInfo.java]: 👍

File: src/test/java/uk/gov/pay/connector/it/base/ChargingITestBase.java (hunk 3 above)
Comment (danworth) [reply src/test/java/uk/gov/pay/connector/it/base/ChargingITestBase.java]: That's a good challenge, I think you're right, I'll remove it.

File: src/main/java/uk/gov/pay/connector/wallets/common/api/PaymentInfo.java (hunk 4 above)
Comment (danworth) [reply src/main/java/uk/gov/pay/connector/wallets/common/api/PaymentInfo.java]: Ok, how about putting it into `wallets/model`?

**Review by georgeracu — APPROVED** [review]
LGTM 👍

**Pull request merged**